// Copyright 2024 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

option java_multiple_files = true;
option java_package = "org.chromium.components.sync.protocol";

option optimize_for = LITE_RUNTIME;

package data_sharing_pb;

import "components/sync/protocol/collaboration_group_metadata.proto";

enum MemberRole {
  MEMBER_ROLE_UNSPECIFIED = 0;
  MEMBER_ROLE_OWNER = 1;
  MEMBER_ROLE_MEMBER = 2;
  MEMBER_ROLE_INVITEE = 3;
  MEMBER_ROLE_FORMER_MEMBER = 4;
};

message GroupMember {
  optional string gaia_id = 1;
  optional string display_name = 2;
  optional string email = 3;
  optional MemberRole role = 4;
  optional string avatar_url = 5;
  optional string given_name = 6;
};

message GroupData {
  optional string group_id = 1;
  optional string display_name = 2;
  repeated GroupMember members = 3;
  optional string access_token = 4;
  repeated GroupMember former_members = 5;
  optional sync_pb.CollaborationGroupMetadata collaboration_group_metadata = 6;
};
